app/src/main/java/com/lagradost/cloudstream3/MainAPI.kt

```diff
@@ -163,6 +163,38 @@ object APIHolder {
         return null
     }
 
+    /**
+     * Get anime tracker information based on title, year and type.
+     * Both titles are attempted to be matched with both Romaji and English title.
+     * Uses the consumet api.
+     *
+     * @param mainTitle Title to search by and match
+     * @param secondaryTitle Optional extra title if you have multiple titles and want extra guarantee to match.
+     * @param types Optional parameter to narrow down the scope to Movies, TV, etc. See TrackerType.getTypes()
+     * @param year Optional parameter to only get anime with a specific year
+     **/
+    suspend fun getTracker(
```

Comment:
make a cached version of this that stores them in a hashmap also make it safe via try
var trackerCache : HashMap<String, AniSearch> = hashMapOf()
suspend fun getTracker(...) : Tracker? {
try {
let mainTitle = titles[0]
let search = trackerCache.get(mainTitle) :? app.get("https://api.consumet.org/meta/anilist/$mainTitle")
            .parsedSafe<AniSearch>()?.also { trackerCache[mainTitle] = it } ?: return null
...
} catch(e : Throwable) {
return null
}
}